### PR TITLE
chore: enforce PR title format with PRLint Reloaded

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Enforce Conventional Commit format of PR title using [PRLint Reloaded](https://github.com/marketplace/prlintreloaded)

Step 1: Allow access to the `codeigniter4` org then to this repo for the app (I have no rights to do so)
Step 2: Merge this PR to let the configuration kick in

For the regex reference: https://regex101.com/r/Yh1iT9/1

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

**Comment by michalsn** [comment]
I'm not sure what others will say, but I believe we were thinking of going in the opposite direction in terms of tightening PR requirements.

Potentially speaking... every commit would be checked, or only the most recent?

**Comment by sfadschm** [comment]
I think the idea of this PR is great, but it will definitely be an additional hurdle especially for beginners to contribute.

However, many PRs consist of numerous commits with unhelpful commit messages.

So my two cents are:
- If PRs are merged into the respective branches "as is", so pulling through every single commit, this approach will greatly increase readability and traceability, but keep new people from contributing to the project.
- If PRs are merged by "squash and merge", the goal would rather be to have the commit message of the squashed commit linted. E.g., make sure that the PR title is useful and then use that as the commit message to be merged.

**Comment by sfadschm** [comment]
This approach only checks PR titles upon creation, not commit messages involved in the PR, correct?

If so, I think this is quite useful and not a hurdle keeping beginners from contributing.

**Comment by michalsn** [comment]
Ok, maybe I didn't fully understand it. If only the PR title is checked, I would be fine with it.

File: .github/prlint.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "title": [
+        {
+            "pattern": "^(\\[\\d+\\.\\d+\\]\\s{1})?(feat|fix|chore|docs|refactor|style|test)(\\([\\-.@:`a-zA-Z0-9]+\\))?!?:\\s{1}\\S.+\\S$",
+            "message": "PR title does not meet requirement, use Conventional Commits format using one of these categories: feat, fix, chore, docs, refactor, style, test"
```
Comment (kenjis) [inline .github/prlint.json:5]: I agree with standardizing PR titles as it makes CHANGELOG.md easier to read.

This message seems difficult to understand what to do for beginners.
The meaning of each category and examples should be shown.

And please add `perf` for performance tuning.

**Review by kenjis — APPROVED** [review]

**Review by lonnieezell — APPROVED** [review]
If I'm understanding how and when this works then I think this is fine. Will also make compiling our changelogs simpler.

**Comment by paulbalandan** [comment]
@lonnieezell thanks for the approval! can you install the Github app to the org so this can be merged?

**Comment by lonnieezell** [comment]
> @lonnieezell thanks for the approval! can you install the Github app to the org so this can be merged?

Sure. How would I do this?

**Comment by paulbalandan** [comment]
Install it from here: https://github.com/marketplace/prlintreloaded

**Comment by lonnieezell** [comment]
@paulbalandan Should be installed now for the codeigniter4 org.

**Comment by paulbalandan** [comment]
Rebased to test the app. And all passed 🎉

**Pull request merged**